src/event-helper.js

```diff
@@ -108,44 +108,52 @@ export function isLoaded(element) {
 export function loadPromise(element, opt_timeout) {
   let unlistenLoad;
   let unlistenError;
-  const loadingPromise = new Promise((resolve, reject) => {
-    if (isLoaded(element)) {
-      resolve(element);
+  if (isLoaded(element)) {
```

Comment:
AMP is designed with the assumption that promises are heavily optimized in browsers, which is slowly becoming actually true.

If that deferred would be implemented in terms of a promise than it uses strictly more memory. You could get rid of the closure (or bound function) passed to the `Promise` constructor by using global variables (shiver), but that might itself be more expensive.